tests/integration_tests/tests/connect_info.rs

```diff
@@ -48,3 +48,79 @@ async fn getting_connect_info() {
 
     jh.await.unwrap();
 }
+
+#[cfg(unix)]
+pub mod unix {
+    use std::path::Path;
+
+    use futures_util::FutureExt;
+    use tokio::{
+        net::{UnixListener, UnixStream},
+        sync::oneshot,
+    };
+    use tokio_stream::wrappers::UnixListenerStream;
+    use tonic::{
+        transport::{
+            server::UdsConnectInfo,
+            Endpoint, Server, Uri,
+        },
+        Request, Response, Status,
+    };
+    use tower::service_fn;
+
+    use integration_tests::pb::{test_client, test_server, Input, Output};
+
+    struct Svc {}
+
+    #[tonic::async_trait]
+    impl test_server::Test for Svc {
+        async fn unary_call(&self, req: Request<Input>) -> Result<Response<Output>, Status> {
+            // Client-side unix domain sockets are unnamed.
+            assert!(req.remote_addr().is_none());
+            assert!(req.extensions().get::<UdsConnectInfo>().is_some());
+
+            Ok(Response::new(Output {}))
+        }
+    }
+
+    #[tokio::test]
+    async fn getting_connect_info() {
+        let unix_socket_path = "/tmp/uds-integration-test";
```

Comment:
Yeah, I was more in the camp of "the cat's already out of the bag". All of the tests here listen on fixed ports.

The current UDS example is even worse [since it makes no effort to clean up the socket at all](https://github.com/StarryInternet/tonic/blob/c62f382e3c6e9c0641decfafb2b8396fe52b6314/examples/src/uds/server.rs#L42) - if you were to run the server, kill it, then run the server again, you'd get an error for the file already existing. Figured it was ok to not go too crazy here but put in _some_ effort to clean up the file as a middle-ground.